CMakeLists.txt

```diff
@@ -387,13 +428,27 @@ target_link_libraries( ${PROJECT_NAME} PUBLIC
 	Qt${QT_VERSION_MAJOR}::PrintSupport
 	Qt${QT_VERSION_MAJOR}::Svg)
 
+if(ENABLE_DIALOG_DESIGNER)
+    target_compile_definitions(${PROJECT_NAME} PUBLIC -DHAVE_QTDESIGNER_INTEGRATION)
+    if(${QT_VERSION_MAJOR} STREQUAL 5)
+        target_link_libraries( ${PROJECT_NAME} PUBLIC
+            Qt${QT_VERSION_MAJOR}::Designer
+            Qt${QT_VERSION_MAJOR}::DesignerComponents)
+    endif()
+
+    if(${QT_VERSION_MAJOR} STREQUAL 6)
+        target_link_libraries( ${PROJECT_NAME} PUBLIC
+            Qt${QT_VERSION_MAJOR}::Designer
+            Qt${QT_VERSION_MAJOR}::DesignerComponentsPrivate)
+    endif()
+endif(ENABLE_DIALOG_DESIGNER)
+
 if(ENABLE_ZINT)
     target_link_libraries( ${PROJECT_NAME} PRIVATE QZint)
 endif(ENABLE_ZINT)
 
 target_compile_definitions( ${PROJECT_NAME} PRIVATE -DHAVE_QT${QT_VERSION_MAJOR} -DHAVE_REPORT_DESIGNER -DUSE_QJSENGINE -D_CRT_SECURE_NO_WARNINGS)
```

Comment:
Look at HAVE_REPORT_DESIGNER
I dont know how it relates to HAVE_QTDESIGNER_INTEGRATION
But linking problem was from https://github.com/fralx/LimeReport/blob/master/limereport/lrfactoryinitializer.cpp#L61 that depends on HAVE_REPORT_DESIGNER while resource dialogdesigner depends on HAVE_QTDESIGNER_INTEGRATION